File: android/app/src/main/java/com/example/checkpoint_app/BootReceiver.java
package com.example.checkpoint_app;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

/**
 * Receiver qui intercepte le démarrage du téléphone (BOOT_COMPLETED).
 * Il permet de relancer automatiquement l'application au démarrage pour assurer le mode Kiosque.
 */
public class BootReceiver extends BroadcastReceiver {
    @Override
    public void onReceive(Context context, Intent intent) {
        if (Intent.ACTION_BOOT_COMPLETED.equals(intent.getAction())) {
            Intent i = new Intent(context, MainActivity.class);
            i.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK);
            context.startActivity(i);
        }
    }
}

File: android/app/src/main/kotlin/com/example/checkpoint_app/BootReceiver.kt
package com.example.checkpoint_app

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent

class BootReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        if (Intent.ACTION_BOOT_COMPLETED == intent.action) {
            val i = Intent(context, MainActivity::class.java)
            i.setClassName(context.packageName, "com.example.checkpoint_app.MainActivity")
            i.addFlags(Intent.FLAG_ACTIVITY_NEW_TASK)
            context.startActivity(i)
        }
    }
}
